go/v4/exchange_eth.go

```diff
@@ -208,37 +208,80 @@ func (this *Exchange) EthAbiEncode(types interface{}, args interface{}) interfac
 	return byteArray
 }
 
+// func ConvertInt64ToBigInt(data interface{}) interface{} {  // these functions change in place the object, no bueno
+// 	switch v := data.(type) {
+// 	case map[string]interface{}:
+// 		for key, value := range v {
+// 			v[key] = ConvertInt64ToBigInt(value)
+// 		}
+// 		return v
+// 	case []interface{}:
+// 		for i, item := range v {
+// 			v[i] = ConvertInt64ToBigInt(item)
+// 		}
+// 		return v
+// 	case int64:
+// 		return uint8(v)
+// 	default:
+// 		return v // Leave other types unchanged
+// 	}
+//
+// }
+
 func ConvertInt64ToBigInt(data interface{}) interface{} {
 	switch v := data.(type) {
 	case map[string]interface{}:
+		newMap := make(map[string]interface{}, len(v))
 		for key, value := range v {
-			v[key] = ConvertInt64ToBigInt(value)
+			newMap[key] = ConvertInt64ToBigInt(value)
 		}
-		return v
+		return newMap
 	case []interface{}:
+		newSlice := make([]interface{}, len(v))
 		for i, item := range v {
-			v[i] = ConvertInt64ToBigInt(item)
+			newSlice[i] = ConvertInt64ToBigInt(item)
 		}
-		return v
+		return newSlice
 	case int64:
 		return uint8(v)
```

Comment:
@carlosmiei I wonder why do we return uint8 here? Should it return BitInt (function name is `ConvertInt64ToBigInt`)?

Reply:
@sc0Vu I have to test things, I don't remember exactly

Reply:
seems to be working now